File: valid_Parentheses.cpp
#include <iostream>
using namespace std;
#include <stack>
class Solution
{
public:
  bool isValid(string s)
  {
    stack<char> sk;
    for (auto i : s)
    {
      switch (i)
      {
      case '{':
      case '(':
      case '[':
        sk.push(i);
        break;
      case '}':
        if (sk.empty() || sk.top() != '{')
        {
          return false;
        }
        else
        {
          sk.pop();
        }
        break;
      case ')':
        if (sk.empty() || sk.top() != '(')
        {
          return false;
        }
        else
        {
          sk.pop();
        }
        break;
      case ']':
        if (sk.empty() || sk.top() != '[')
        {
          return false;
        }
        else
        {
          sk.pop();
        }
        break;
      default:
        return false;
        break;
      }
    }
    return sk.empty();
  }
};

File: ValidParentheses.py
# this is the codeWars version (won't work on leetcode
# as of different problem requirements )

from collections import deque


def valid_parentheses(string):
    stack = deque()
    s = len(string)
    for i in range(s):
        if string[i] == "(":
            stack.append(string[i])
        else:
            if string[i] != "(" and string[i] != ")":
                continue
            if not stack:  # checks if the stack is empty
                return False
            else:
                stack.pop()

    if stack:
        return False
    else:
        return True
